### DISCUSSION TITLE
Welcome to Jacdac

### DISCUSSION THREAD

**Discussion by pelikhan** [microsoft/jacdac#126] [General]
Welcome the Jacdac community forum!

Please use these discussions to ask anything about Jacdac!

**Comment by manubrico** [comment]
Hi,
I need more information about hardware design. I would like to use the JACDAC connector for my design. I see that use UART and GPIO, do you have hardware design example? 
Thank you

**Reply by jamesadevine** [reply]
Hiya! welcome :smile:

We will be releasing schematics soon.

What do you intend to use Jacdac for?

**Reply by manubrico** [reply]
Hi,
Thank for reply
I would like to use with STM or nRF µP but I need more information about hardware design.
I saw in documentation that use one UART and one GPIO. UART will be for TX and RX in the same line because JACDAC use only 3 PIN (VCC, GND, and DATA), I suppose that GPIO  change the hardware direction RX or TX.
Do you have some information about that?

**Reply by pelikhan** [reply]
is µP short for micropython?

**Reply by manubrico** [reply]
No just  micro controller abbreviation.

**Reply by jamesadevine** [reply]
We have documentation in the form of firmware if that helps?

https://github.com/microsoft/jacdac-stm32x0/blob/main/stm32/uart.c#L229
https://github.com/microsoft/pxt-jacdac/blob/master/jdlow.c

Modern microcontrollers allow us to reuse the same GPIO for interrupts and for UART TX/RX, so we typically use just one pin.

**Reply by jamesadevine** [reply]
also, this: https://microsoft.github.io/jacdac-docs/reference/protocol/#single-wire-serial-sws

**Reply by pelikhan** [reply]
@manubrico it took a little longer than expected but the device development kit is now complete https://microsoft.github.io/jacdac-docs/ddk/
